src/linear_algebra/matrix/nonsingular_inverse.lean

```diff
@@ -36,11 +57,119 @@ open equiv equiv.perm finset
 
 variables (A : matrix n n α) (B : matrix n n α)
 
+/-! ### Matrices are `invertible` iff their determinants are -/
+
+section invertible
+
+/-- A copy of `inv_of_mul_self` using `⬝` not `*`. -/
+protected lemma inv_of_mul_self [invertible A] : ⅟A ⬝ A = 1 := inv_of_mul_self A
+
+/-- A copy of `mul_inv_of_self` using `⬝` not `*`. -/
+protected lemma mul_inv_of_self [invertible A] : A ⬝ ⅟A = 1 := mul_inv_of_self A
+
+/-- If `A.det` has a constructive inverse, produce one for `A`. -/
+def invertible_of_det_invertible [invertible A.det] : invertible A :=
+{ inv_of := ⅟A.det • A.adjugate,
+  mul_inv_of_self :=
+    by rw [mul_smul_comm, matrix.mul_eq_mul, mul_adjugate, smul_smul, inv_of_mul_self, one_smul],
+  inv_of_mul_self :=
+    by rw [smul_mul_assoc, matrix.mul_eq_mul, adjugate_mul, smul_smul, inv_of_mul_self, one_smul] }
+
+lemma inv_of_eq [invertible A.det] [invertible A] : ⅟A = ⅟A.det • A.adjugate :=
+begin
+  letI := invertible_of_det_invertible A,
+  have : ⅟A = ⅟A.det • A.adjugate := rfl,
+  convert this,
+end
```

Comment:
This lemma is new. I couldn't work out how to golf the last two lines into a single line

Reply:
```suggestion
lemma inv_of_eq [invertible A.det] [invertible A] : ⅟A = ⅟A.det • A.adjugate :=
by { letI := invertible_of_det_invertible A, convert (rfl : ⅟A = ⅟A.det • A.adjugate) }
```